KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,652 @@
+                return i;
+            }
+        }
+        revert("Retired does not exist");
+    }
+
+    /**
+     * @dev to calculate the sum of retirees balances
+     * @return retireesBalance the sum of balances of retireds
+     */
+    function sumOfRetiredBalance() public view returns (uint256 retireesBalance) {
+        for (uint8 i = 0; i < retirees.length; i++) {
+            address retiredAddress = retirees[i].retired;
+            retireesBalance += retiredAddress.balance;
+        }
+        return retireesBalance;
+    }
+
+    /**
+     * @dev to get newbie details by newbieAddress
+     * @param _newbieAddress is the address of the newbie
+     * @return newbie is the address of the newbie
+     * @return amount is the fund allocated to the newbie
+
+     */
+    function getNewbie(
+        address _newbieAddress
+    ) public view returns (address, uint256) {
+        require(newbieExists(_newbieAddress), "Newbie does not exist");
+        uint256 index = getNewbieIndex(_newbieAddress);
+        Newbie memory newbie = newbies[index];
+        return (newbie.newbie, newbie.amount);
+    }
+
+    /**
+     * @dev check whether _newbieAddress is registered
+     * @param _newbieAddress is the address of the newbie
+     */
+    function newbieExists(
+        address _newbieAddress
+    ) public view returns (bool) {
+        require(_newbieAddress != address(0), "Invalid address");
+        for (uint8 i = 0; i < newbies.length; i++) {
+            if (newbies[i].newbie == _newbieAddress) {
+                return true;
+            }
+        }
+    }
+
+    /**
+     * @dev get index of the newbie in the newbies array
+     * @param _newbieAddress is the address of the newbie
+     */
+    function getNewbieIndex(
+        address _newbieAddress
+    ) public view returns (uint256) {
+        for (uint256 i = 0; i < newbies.length; i++) {
+            if (newbies[i].newbie == _newbieAddress) {
+                return i;
+            }
+        }
+        revert("Newbie does not exist");
+    }
+
+    /**
+     * @dev to calculate the sum of newbie funds
+     * @return treasuryAmount the sum of funds allocated to newbies
+     */
+    function getTreasuryAmount() public view returns (uint256 treasuryAmount) {
+        for (uint8 i = 0; i < newbies.length; i++) {
+            treasuryAmount += newbies[i].amount;
+        }
+        return treasuryAmount;
+    }
+
+    /**
+     * @dev gets the length of retirees list
+     */
+    function getRetiredCount() public view returns (uint256) {
+        return retirees.length;
+    }
+
+    /**
+     * @dev gets the length of newbies list
+     */
+    function getNewbieCount() public view returns (uint256) {
+        return newbies.length;
+    }
+
+    /**
+     * @dev allback function to revert any payments
+     */
+    fallback() external payable {
+        revert("This contract does not accept any payments");
+    }
+
```

Comment:
it would be good to have a function that returns all the retired and newbie addresses when the status is finalized. It is useful for ecosystem service providers like scope and finder.

Reply:
like getting the senders and receivers in the txn receipt ?
```
function finalizeContract( string memory _memo)
        public
        onlyOwner
        onlyAtStatus(Status.Approved)
        returns (address[] memory retirees, address[] memory newbies)
 ```

Reply:
yes, with the balances.

Reply:
added

Reply:
Ah, sorry for that. I misunderstood. My intention is that it could be good to have a view function that anyone can call this contract to get the all information about the retirees and newbies. If we just make a receipt, it is not easy to read. To get the value afterwards, we need to know the exact block number to get the receipt.

Reply:
We have separate getter function for retired and newbie 
https://github.com/klaytn/treasury-rebalance/blob/main/contracts/TreasuryRebalance.sol#L370. 

Do we need one function to view all the values like ?
```
function getSnapshot() public view returns(
            address[] memory retirees, 
            uint256 totalRetireesBalance,
            address[] memory newbies,
            uint256 totalNewbiesFund) 
 {
 }
 ```

Reply:
@toniya-klaytn Yes, this is the one that I wanted. I haven't seen that before.